### PR TITLE
Caldav write

### PR DESCRIPTION
**Pull request by voidus**
http://jira.opensource.mayflower.de/jira/browse/PHPROJEKT-233

Basic create/update/delete support for thunderbird.

### CODE REVIEW

**Comment by voidus** [comment]
https://github.com/Mayflower/PHProjekt/pull/45 should be merged before this

**Comment by voidus** [comment]
This uses https://github.com/Mayflower/PHProjekt/pull/56, so it should be merged before this.

File: phprojekt/application/Calendar2/CalDAV/CalendarBackend.php [hunk 1]
```diff
@@ -130,16 +130,41 @@ class Calendar2_CalDAV_CalendarBackend extends Sabre_CalDAV_Backend_Abstract
 
     public function createCalendarObject($calendarId, $objectUri, $calendarData)
     {
-        throw new Exception('Calendar2_CalDAV_CalendarBackend->createCalendarObject is not implemented yet');
+        $vcalendar = Sabre_VObject_Reader::read($calendarData);
+        $event = new Calendar2_Models_Calendar2();
```
Comment (exi) [inline phprojekt/application/Calendar2/CalDAV/CalendarBackend.php]: indention

File: phprojekt/application/Calendar2/Models/Calendar2.php [hunk 2]
```diff
@@ -940,6 +936,91 @@ class Calendar2_Models_Calendar2 extends Phprojekt_Item_Abstract
...
+        // none or one of these two
+        //  dtend, duration (only assumes dtend case for now)
+        //
+        // 0 - n
+        // TODO: Check how we can handle these. Maybe just concat them?
+        // handling: (only one is handled atm, though)
+        //  comment, rrule
+        // not handling:
+        //  attach, attendee, categories, contact, exdate, exrule, rstatus, related, resources, rdate, x-prop
+        $utc               = new DateTimezone('UTC');
```
Comment (exi) [inline phprojekt/application/Calendar2/Models/Calendar2.php]: indention

File: phprojekt/application/Calendar2/Models/Calendar2.php [hunk 3]
```diff
@@ -940,6 +936,91 @@ class Calendar2_Models_Calendar2 extends Phprojekt_Item_Abstract
...
+        $start->setTimezone($utc);
+        $this->start = $start->format('Y-m-d H:i:s');
+
+        if (substr($vevent->dtend->value, -1) === 'Z') {
+            $end = new Datetime($vevent->dtend->value);
+        } else {
+            $end = new Datetime($vevent->dtend->value, new DateTimezone($vevent->dtend['tzid']->value));
+        }
+        $end->setTimezone($utc);
+        $this->end   = $end->format('Y-m-d H:i:s');
```
Comment (exi) [inline phprojekt/application/Calendar2/Models/Calendar2.php]: indention

**Pull request merged**